osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs

```diff
@@ -331,6 +332,7 @@ private void handleTimingPoints(Beatmap beatmap, string line)
 
             if (speedMultiplier != difficultyPoint.SpeedMultiplier)
             {
+                beatmap.ControlPointInfo.DifficultyPoints.RemoveAll(x => x.Time == time);

```

Comment:
I explicitly did only the difficulty points, because it's reset to 1 with a regular timing point.
If there are no negative side effects, I'll strip out duplicates afterward instead.